hoomd/mpcd/test/rejection_sphere_geometry_filler_test.cc

```diff
@@ -0,0 +1,156 @@
+            // type should be set
+            UP_ASSERT_EQUAL(__scalar_as_int(h_pos.data[i].w), 1);
+
+            Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > Scalar(25))
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        }
+
+    /*
+     * Fill the volume again, which should approximately double the number of virtual particles
+     */
+    filler->fill(1);
+    std::cout << "Number of virtual Particles after filling twice: " << pdata->getNVirtual() << "\n";
+        {
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+        ArrayHandle<unsigned int> h_tag(pdata->getTags(), access_location::host, access_mode::read);
+
+        // check if the particles have been placed outside the confinement
+        unsigned int N_out(0);
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            // tag should equal index on one rank with one filler
+            UP_ASSERT_EQUAL(h_tag.data[i], i);
+            // type should be set
+            UP_ASSERT_EQUAL(__scalar_as_int(h_pos.data[i].w), 1);
+
+            Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > Scalar(25))
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        }
+
+    /*
+     * Test the average properties of the virtual particles.
+     */
+    // initialize variables for storing avg data
+    unsigned int N_avg(0);
+    Scalar3 vel_avg = make_scalar3(0,0,0);
+    Scalar T_avg(0);
+    // repeat filling 100 times
+    for (unsigned int t=0; t<100; ++t)
+        {
+        pdata->removeVirtualParticles();
+        filler->fill(2+t);
+
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+
+        // local variables
+        unsigned int N_out(0);
+        Scalar temp(0);
+
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            const Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar3 vel = make_scalar3(h_vel.data[i].x, h_vel.data[i].y, h_vel.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > Scalar(25))
+                ++N_out;
+            temp += dot(vel, vel);
+            vel_avg += vel;
+            }
+
+        temp /= (3*(N_out-1));
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        N_avg += N_out;
+        T_avg += temp;
+
+        }
+    vel_avg /= N_avg;
```

Comment:
Don't you want to divide `N_avg` by the number of frames here too? Otherwise the check below becomes confusing if someone decides to increase the number of samples at a later date.

Reply:
Hmm. I guess, I have done a mistake here. I did not divide the velocity by the number of particles in the inner loop. I will correct that. 
I was dividing by the total number of virtual particles added in all the iterations. but it is not the same.
Thanks for pointing this out.

Reply:
This was actually more or less OK before because you did not divide `N_avg` or `vel_avg` by the number of samples (100), but I agree that I would do it that way instead.

Reply:
The "more or less" was referring to this point =)

Reply:
Ah ok. :)